README.rst

```diff
@@ -69,6 +69,168 @@ The *apidoc* extension uses the following configuration values:
+  document. As a result, it is necessary to remove
+  ``autodoc_tree_index_modules`` from ``setup.cfg`` and define
+  ``apidoc_module_dir`` in ``conf.py``.
+
+- Replace ``autodoc_tree_excludes`` with ``apidoc_excluded_paths``
+
+  Like ``apidoc_excluded_paths``, ``autodoc_tree_excludes`` is a list of
+  fnmatch-style paths describing files and directories to exclude relative to
+  the source directory. This means it's a like-for-like replacement so you can
+  simply move the option to ``conf.py``, update the paths (assuming
+  ``apidoc_module_dir`` is configured to anything except ``.``), and rename it.
+
+- Replace ``api_doc_dir`` with ``apidoc_output_dir``
+
+  As above, ``api_doc_dir`` functions exactly as ``apidoc_output_dir`` does.
+  Simply move the option to ``conf.py`` and rename it.
+
+Once done, your output should work exactly as before.
+
+autodoc
+~~~~~~~
+
+*autodoc* is not based on *sphinx-apidoc*. Fortunately it is possible to
+generate something very similar (although not identical!). Let's take
+`oslo.privsep`_ as an example, once again looking at minimal versions of
+``setup.cfg`` and ``doc/source/conf.py``:
+
+.. code-block:: ini
+
+   [build_sphinx]
+   all_files = 1
+   build-dir = doc/build
+   source-dir = doc/source
+
+   [pbr]
+   autodoc_index_modules = True
+   api_doc_dir = reference/api
+   autodoc_exclude_modules =
+     oslo_privsep.tests.*
+     oslo_privsep._*
+
+.. code-block:: python
+
+   extensions = ['']
+
+Once migrated, this would look like so:
+
+.. code-block:: ini
+
+   [build_sphinx]
+   all_files = 1
+   build-dir = doc/build
+   source-dir = doc/source
+
+.. code-block:: python
+
+   extensions = ['sphinxcontrib.apidoc']
+
+   apidoc_module_dir = '../../oslo_privsep'
+   apidoc_excluded_paths = ['tests', '_*']
+   apidoc_output_dir = 'reference/api'
+   apidoc_separate_modules = True
+
+Most of the changes necessary are the same as `autodoc_tree`_, with some
+exceptions.
+
+- Replace ``autodoc_index_modules`` with ``apidoc_module_dir``
+
+  Same as the step required for *autodoc_tree*, but you're removing
+  ``autodoc_index_modules`` instead of ``autodoc_tree_index_modules``.
+
+- Replace ``autodoc_tree_excludes`` with ``apidoc_excluded_paths``
+
+  ``autodoc_exclude_modules`` differs from ``apidoc_excluded_paths`` and
+  ``autodoc_tree_excludes`` in that it's a list of fnmatch-style **module
+  paths** - not file paths. As a result, you must switch from ``x.y`` format to
+  ``x/y``. These paths should be relative to ``apidoc_module_dir``, not
+  ``setup.cfg`` as before.
+
+- Replace ``api_doc_dir`` with ``apidoc_output_dir``
+
+  Same as the step required for *autodoc_tree*.
+
+- Configure ``apidoc_separate_modules=True``
+
+  By default, *sphinx-apidoc* generates a document per package while *autodoc*
+  generates a document per (sub)module. By setting this attribute to ``True``,
+  we ensure the latter behavior is used.
+
```

Comment:
Should also mention that you need to replace any links to the file `<apidoc_output_dir>/autoindex.rst`. The file `<apidoc_output_dir>/modules.rst` contains a (hierarchical) index of the documented modules that may be useful as a replacement, but there is no direct equivalent to the flat list in the `autoindex` file.

Reply:
+1

Reply:
Done